Automattic/wp-calypso
# Hosting Command Palette: Change the site order on Calypso based on /sites sort algorithm

Closes https://github.com/Automattic/dotcom-forge/issues/4650

## Proposed Changes

This PR adds the sort to the sites in the nested commands so that the site order is consistent with whatever order is set on `/sites` page. 

Here is a quick demo:

https://github.com/Automattic/wp-calypso/assets/25575134/789b8777-4359-497b-b273-c3049209a898

## Testing Instructions

* Navigate to Calypso Live link
* Navigate to `/sites` page
* Click `cmd + k` on macOS or `ctrl + k` on Windows to trigger command palette
* Select one of the commands that has a nested command e.g. a command with a chevron on the right - `Open site dashboard` would be a good command to test as it includes all sites and does not filter any sites
* Observe the list of sites and confirm its order is consistent with the list of sites displayed on `/sites` page
* Exit the command palette
* Change the sort type on the `/sites` page to a different type than previously selected:

<img width="1263" alt="Screenshot 2023-11-29 at 9 17 06 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/45fc2025-fdbd-4399-8faa-4843cc56ea07">

* Click `cmd + k` on macOS or `ctrl + k` on Windows to trigger command palette
* Select one of the commands that has a nested command e.g. a command with a chevron on the right
* Confirm that the sort order was changed and is consistent with the `/sites` page


## Pre-merge Checklist
